updated docs (#362)
ratschlab/metagraph: metagraph/docs/source/quick_start.rst

.. TODO: note that canonical graphs must not be used with row-diff<*> annotations and always must be primarized

Graph cleaning
^^^^^^^^^^^^^^

For removing sequencing errors, MetaGraph provides routines for graph cleaning and k-mer
filtering. These are based on the assumption that

> k-mers with relatively low abundance (k-mer counts) in the input data are likely due to sequencing errors, and

Reviewer: ```suggestion
k-mers with relatively low abundance (k-mer counts) in the input data likely represent sequencing errors, and
```

> hence should be dropped to keep the graph free of the non-existent k-mers.

Reviewer: ```suggestion
hence should be dropped to reduce noise in the final graph.
```
The previous statement was too strong, we never guarantee that our cleaning strategy is perfect.

Author: Strictly speaking, the new statement is wrong too.

Well, this is not a theorem, anyway.

Author: I don't like that it says there is noise in the graph. There is no noise. It contains all k-mers from the input.

Reviewer: It depends on what you choose as your baseline: 1) the true DNA sequence of the individual you wanted to sequence? 2) the true DNA sequences of whatever DNA is in your sample?, or 3) the measured k-mers from the sequencing experiment.

Maybe we should talk about this a bit. Currently, the term "non-existent" is not defined.

Reviewer: When you say that there is no noise in the graph, then it sounds like you're considering 3) to be your baseline, in which case, this procedure is removing signal.

Author: It's a curation algorithm that operates on the input graph. There is no ambiguity. Ok, both statements were inaccurate. I'll write something different.

.. TODO: note that canonical graphs must not be used with row-diff<*> annotations and always must be primarized

Graph cleaning
^^^^^^^^^^^^^^

For removing sequencing errors, MetaGraph provides routines for graph cleaning and k-mer
filtering. These are based on the assumption that
k-mers with relatively low abundance (low k-mer counts) in the input data were likely generated due to sequencing errors, and
hence should be dropped. Moreover, to make the cleaning procedure more robust, the decision about filtering out a k-mer can be

> made based on the median abundance of the unitig this k-mer belongs to. That is, k-mers with low abundance are preserved if

Reviewer 2: "can be made based"  --> "can be based"

filtering. These are based on the assumption that
k-mers with relatively low abundance (low k-mer counts) in the input data were likely generated due to sequencing errors, and
hence should be dropped. Moreover, to make the cleaning procedure more robust, the decision about filtering out a k-mer can be

> based on the median abundance of the unitig this k-mer belongs to. That is, k-mers with low abundance are preserved if

Reviewer: ```suggestion
based on the median abundance of the unitig to which this k-mer belongs. That is, k-mers with low abundance are preserved if
```
minor grammar nit....

    zless SRR403017_clean_contigs.fasta.gz

.. note:: The default parameters in ``metagraph clean`` correspond to no cleaning. That is, an equivalent of ``metagraph transform --to-fasta``, which extracts from the input de Bruijn graph contigs including all its k-mers.

For cleaning graphs constructed from high-throughput Illumina reads, the recommended parameters are
``--prune-tips <2k> --prune-unitigs 0 --fallback 2``, which implements the cleaning procedure proposed in `McCortex <https://github.com/mcveanlab/mccortex>`_ (Turner et al., 2018) and includes the following steps:

1. Prune all tips shorter than *2k*, where *k* is the k-mer length.

> 2. Compute a threshold for the minimum k-mer abundance as follows. Assume the number of k-mers with sequencing errors (erroneous k-mers) follows a Poisson distribution with a Gamma distributed mean. Also assume that all k-mers with abundance 3 or less are due to sequencing errors. Based on these numbers, fit a Poisson distribution and pick the threshold such that the estimated erroneous k-mers make up at most 0.1% of the total k-mer coverage at that abundance level. If the picked threshold keeps less than 20% of the total coverage, deem the automatic estimation procedure unsuccessful and use the fallback value of 2 instead (set by flag ``--fallback``).

Reviewer: ```suggestion
2. Compute a threshold for the minimum k-mer abundance as follows. Assume that the number of k-mers with sequencing errors (erroneous k-mers) follows a Poisson distribution with a Gamma-distributed mean. Also, assume that all k-mers with abundance 3 result from sequencing errors. Based on these numbers, fit a Poisson distribution and pick a threshold such that k-mers predicted to be erroneous make up at most 0.1% of the total k-mer coverage at that abundance level. If the chosen threshold keeps less than 20% of the total coverage, deem the automatic estimation procedure unsuccessful and use the fallback value of 2 instead (set by flag ``--fallback``).
```

> 3. Traverse the graph (where all short tips have already been removed in step 1) and fetch all unitigs with the median k-mer abundance greater or equal to the threshold defined in step 2.

Reviewer: ```suggestion
3. Traverse the graph (where all short tips have already been removed in step 1) and fetch all unitigs with a median k-mer abundance greater or equal to the threshold defined in step 2.
```

> Once clean contigs (unitigs) are extracted from a de Bruijn graph, construct a clean de Bruijn graph from them.

Reviewer: ```suggestion
Once all contigs (unitigs of the cleaned graph) are extracted from a de Bruijn graph, construct a clean de Bruijn graph from them.
```

> .. tip:: In case of indexing multiple read sets, the recommended workflow is to build a *sample de Bruijn graph* from each read set separately and clean these sample graphs independently (that is, extract clean contigs from each of them). Next, build a joint de Bruijn graph from all these clean contigs and finally annotate it using the generated clean contig sets instead of the original raw read sets.

Reviewer: ```suggestion
.. tip:: When indexing multiple read sets, the recommended workflow is to build a *sample de Bruijn graph* from each read set separately and clean these sample graphs independently (that is, extract clean contigs from each of them). Next, build a joint de Bruijn graph from all clean contigs and finally annotate it using the generated clean contig sets instead of the original raw read sets.
```

    zless SRR403017_clean_contigs.fasta.gz

.. note:: The default parameters in ``metagraph clean`` correspond to no cleaning. That is, an equivalent of ``metagraph transform --to-fasta``, which extracts from the input de Bruijn graph contigs including all its k-mers.

> 

Reviewer 2: "which extracts from the input de Bruijn graph contigs including all its k-mers." --> "which extracts from the input de Bruijn graph all contigs, without removing any k-mers."
